kong/plugins/oauth2/access.lua

```diff
@@ -92,10 +92,15 @@ local function get_redirect_uri(client_id)
 end
 
 local function retrieve_parameters()
-  ngx.req.read_body()
-
   -- OAuth2 parameters could be in both the querystring or body
-  return utils.table_merge(ngx.req.get_uri_args(), public_utils.get_body_args())
+  local uri_args = ngx.req.get_uri_args()
+  if ngx.req.get_method() ~= "GET" then
+    ngx.req.read_body()
+    local body_args = public_utils.get_body_args()
+    return utils.table_merge(uri_args, body_args)
+  else
```

Comment:
There is no need for this `else` branch if the previous one is guaranteed to `return`. This would be better written as:

```lua
if get_method() ~= "GET" then
  read_body()
  -- ...
  return ...
end

return uri_args
```

Reply:
"else" does not imply a separate "branch" is created? there is no semantically meaningful difference here

Reply:
See https://github.com/Kong/kong/blob/master/CONTRIBUTING.md#conditional-expressions (5th code block)

There is no need for the indentation/boilerplate.

Reply:
(if there are readability concerns, thats certainly valid. just noting that functionally there is no difference)

Reply:
This is indeed about readability.

Reply:
YES, I'll read the CONTRIBUTING.md again.